src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -7,8 +7,32 @@
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
-    @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException(
+                    "Login cannot be null");
+        }
+
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException(
+                    "User with this login already exists");
+        }
+
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException(
+                    "Login must be at least 6 characters long");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RegistrationException(
+                    "Password must be at least 6 characters long");
+        }
```

Comment:
Split this condition into two separate `if` statements. One for checking if password is null and another one for checking password length. This way, you can provide more specific error messages, which will be helpful for the user.